platform/mbed_retarget.cpp

```diff
@@ -904,76 +904,59 @@ extern "C" long PREFIX(_flen)(FILEHANDLE fh)
     return size;
 }
 
-#if defined(TARGET_NUVOTON) || defined(TARGET_RZ_A1XX)
-extern char Image$$ARM_LIB_HEAP$$ZI$$Base[];
-extern char Image$$ARM_LIB_STACK$$ZI$$Base[];
-extern char Image$$ARM_LIB_HEAP$$ZI$$Limit[];
-#define HEAP_BASE (Image$$ARM_LIB_HEAP$$ZI$$Base)
-#define HEAP_LIMIT (Image$$ARM_LIB_HEAP$$ZI$$Limit)
-#define STACK_BASE (Image$$ARM_LIB_STACK$$ZI$$Base)
+#if defined (__ARMCC_VERSION) && (__ARMCC_VERSION >= 6010050)
+__asm(".global __use_two_region_memory\n\t");
+__asm(".global __use_no_semihosting\n\t");
 #else
-extern char Image$$RW_IRAM1$$ZI$$Limit[];
-extern char Image$$ARM_LIB_STACK$$ZI$$Base[];
-#define HEAP_BASE (Image$$RW_IRAM1$$ZI$$Limit)
-#define HEAP_LIMIT (Image$$ARM_LIB_STACK$$ZI$$Base)
-#define STACK_BASE (Image$$ARM_LIB_STACK$$ZI$$Base)
+#pragma import(__use_two_region_memory)
 #endif
 
+extern uint32_t Image$$ARM_LIB_STACK$$ZI$$Length[];
+extern uint32_t Image$$ARM_LIB_STACK$$ZI$$Base[];
+
+extern uint32_t Image$$RW_IRAM1$$ZI$$Limit[];
+
+extern uint32_t Image$$ARM_LIB_HEAP$$ZI$$Base[];
+extern uint32_t Image$$ARM_LIB_HEAP$$ZI$$Length[];
+
+#if !defined(ISR_STACK_SIZE)
+#define ISR_STACK_START       Image$$ARM_LIB_STACK$$ZI$$Base
+#define ISR_STACK_SIZE        Image$$ARM_LIB_STACK$$ZI$$Length
+#endif
+
+#if !defined(HEAP_START)
+#if defined(Image$$ARM_LIB_HEAP$$ZI$$Base) && defined(Image$$ARM_LIB_HEAP$$ZI$$Length)
```

Comment:
@deepikabhavnani `Image$$ARM_LIB_HEAP$$ZI$$Base` and `Image$$ARM_LIB_HEAP$$ZI$$Length` are linker-generated symbols. Can they be used for preprocessor defines?

Reply:
I have not tested it, yes you are right linker symbols will not be valid during pre-processing step... will have to find another condition or define heap_start in target specific file mbed_rtx.h as it was done earlier